TSPL.docc/ReferenceManual/Attributes.md

```diff
@@ -210,8 +210,77 @@ including important milestones.
   obsoleted: <#version number#>
   ```
   The *version number* consists of one to three positive integers, separated by periods.
+
+- The `noasync` argument indicates that the declared symbol can't be used directly
+  in an asynchronous context.
+
+  Because Swift concurrency can resume on a different thread
+  after a potential suspension point,
+  using elements like thread-local storage, locks, mutexes, or semaphores
+  across suspension points can lead to incorrect results.
+
+  To avoid any errors, add an `@available(*, noasync)` attribute to the symbol's declaration:
+
+  ```swift
+  extension pthread_mutex_t {
+    
+    @available(*, noasync)
+    mutating func lock() {
+        pthread_mutex_lock(&self)
+    }
+    
+    @available(*, noasync)
+    mutating func unlock() {
+        pthread_mutex_unlock(&self)
+    }
+  }
+  ```
+ 
+  This attribute tells the compiler to raise a build error 
```

Comment:
```suggestion
  This attribute tells the compiler to report an error 
```

Elsewhere in TSPL, we call these "compile-time errors"; the usual verb is "raise", although some places we have "results in a compile-time error".